web/post-restoration-transition-application/app/utils/compare.ts

```diff
@@ -0,0 +1,17 @@
+export const compare = function (a: object, b: object): boolean {
+  const keysToCompare = Object.keys({ ...a, ...b })
+    .filter(key => ![null, undefined]
+    .includes(a[key]) || ![null, undefined].includes(b[key]))
+  let rv = true
+  for (let i = 0; i < keysToCompare.length; i++) {
```

Comment:
```suggestion
  for (const key of keysToCompare) {
```
and then `a[key]` / `b[key]` for readability